core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerClient.java

```diff
@@ -91,7 +87,8 @@ public static BlockWorkerClient create(UserState userState, GrpcServerAddress ad
    * @param responseObserver the stream observer for the server response
    * @return the stream observer for the client request
    */
-  StreamObserver<ReadRequest> readBlock(StreamObserver<ReadResponse> responseObserver);
+  StreamObserver<ReadRequest> readBlock(StreamObserver<ReadResponse> responseObserver)
+          throws IOException;
```

Comment:
The `BlockWorker` interface is modified to throw `IOException` that comes with using `AbstractClient#retryRPC`

Reply:
feel like we should update the retryRPC signature so we just throw runtime exception

Reply:
In that case, AlluxioStatusException makes more sense here and we can add more information like retryable to AlluxioStatusException